libguard/guard_exception.hpp

```diff
@@ -0,0 +1,100 @@
+#pragma once
+
+#include <exception>
+#include <string>
+
+class guardFileOpen : public std::exception
+{
+  public:
+    explicit guardFileOpen(const std::string& message) : message(message)
+    {
+    }
+
+    virtual const char* what() const noexcept override
+    {
+        return message.c_str();
+    }
+
```

Comment:
Do you think adding the below function will help more to catcher?
- name() - it can return the name of the exception? "GuardFileOpen"?
- description() - it can return the description of the exception? 
- what() - it can return both i.e name and description?

Reply:
I am unable to get what name i.e. function name will do at user end.

Reply:
what extra actions are they planning to do as a part of exception handling? it makes sense only if there is a common action that needs to be done for all types of exception. and then there are extra specific actions that needs to be done for each exception.

Reply:
>if there is a common action that needs to be done for all types of exception. and then there are extra specific actions that needs to be done for each exception.

Yes but, as a library, we can not that restriction - The caller can do anything if he wants so our library supports it. 
If you are planning to do when someone requesting then I am fine - Just giving concern as Reviewer from library context.

Reply:
Will do when required.

Reply:
Created issue https://github.com/ibm-openbmc/dev/issues/3325

Reply:
please create issue in guard repository